### ISSUE TITLE
unique annotateLables

### ISSUE THREAD

**Issue by luna2761** [FVANCOP/ChartNew.js#299]
Hi!

Great library - ChartNew.js. I think it's really useful.
I'm creating line chart, but i wonder if there is an opportunity to create unique annotateLabels? I mean, to create unique tooltip (annotateLabel) for each point.

Thanks!

**Comment by FVANCOP** [comment]
Not sure that I understand what you mean. Perhaps, is the option "detectAnnotateOnFullLine:true" the answer to your issue  ( see https://github.com/FVANCOP/ChartNew.js/wiki/100_020_Annotations#detectannotateonfullline and/or issue  https://github.com/FVANCOP/ChartNew.js/issues/211)

**Comment by luna2761** [comment]
I mean, instead of "pSecond graph - January:0.65" will be custom text, kind of "bla-bla...", inique for each point.

**Comment by luna2761** [comment]
here is what I mean:
![image](https://cloud.githubusercontent.com/assets/10071245/7980815/4d69f89c-0ab3-11e5-9299-594c581608b1.jpg)

**Comment by FVANCOP** [comment]
For the fixed text, you can specify a function for option annotateLabel (see https://github.com/FVANCOP/ChartNew.js/wiki/085_option_array_func) : 

annotateLabel: myFunctionAnnotate
(...)
function myFunctionAnnotate(area, ctx, data,statData, posi,posj,othervars) {
var tooltip=[];
tooltip[0]=["text1", "hey", "hello", "123", "456", "ttt", "fgg"];
tooltip[1]=["text2", "hey2", "hello2", "hi", "go", "yes", "yohooo"];
return("<%='"+tooltip[posi][posj]+"'%>");
}



For the position : the position is always related to the mouse position; It is not possible to put the annotate on a fixed position (unless you rewrite the function "doMouseFunction").

**Comment by FVANCOP** [comment]
I'm back with another very much simpler solution for the fixed text and with a solution for the fixed position :-)

As you proposed, put the tooltip variable in the data and specifies the following value for the annotateLabel option :  
       annotateLabel : "<%=data.datasets[i].tooltip[j]%>" 
       

var mydata1 = {
labels: ["January", "February", "March", "April", "May", "June", "July"],
datasets: [
{
fillColor: "rgba(220,220,220,0.5)",
strokeColor: "rgba(220,220,220,1)",
pointColor: "rgba(220,220,220,1)",
pointStrokeColor: "#fff",
data: [0.65, 0.59, 0.90, 0.81, 0.56, 0.55, 0.40],
tooltip : ["text1", "hey", "hello", "123", "456", "ttt", "fgg"],
title: "pFirst data"
},
{
fillColor: "rgba(151,187,205,0.5)",
strokeColor: "rgba(151,187,205,1)",
pointColor: "rgba(151,187,205,1)",
pointStrokeColor: "#fff",
data: [0.28, 0.48, 0.40, 0.19, 0.96, 0.27, 1.00],
tooltip : ["text2", "hey2", "hello2", "hi", "go", "yes", "yohooo"],
title: "pSecond data"
}
]
}


For the fixed position : 

Update the function "b_moveIt" in your JS file after the include of ChartNew.js to something like this (change value for xb and yb to your own defined position in the Chart).

<SCRIPT>
function b_moveIt(x, y) {
	var xb=600;
	var yb=900;
	this.x = xb;
	this.y = yb;
	this.css.left = this.x + "px";
	this.css.top = this.y + "px";
};
</SCRIPT>;

**Comment by luna2761** [comment]
No,no, I didn't need fixed position. Just fixed unique text =). But this is an interesting opportunity, too.  So simple... I need to study arrays better)))).

Thank you very very much!!!

**Comment by luna2761** [comment]
Sorry for reopenning, have another question cocerning annotateLabel.
Is there opportunity to move over the hint with mouse (over annotateLaabel)?

Example with css code:
style:
a {
    position:relative;
}
a span
{
  display:none;
}
a:hover span
{
  position:absolute;
  display:inline;
  border: 1px solid #0000FF;
  margin: 0px 0px 0px 10px;
  background: #faffc3;
  right:0;
  margin-right: 105%;
}

html:
<div>
  Some text, 
    <a href="#">link<span>Hint.<br/>Second line of hint.</span></a>
    , some text.
</div>

First, I've tried to begin with adding styles through annotateClassName (https://github.com/FVANCOP/ChartNew.js/wiki/100_020_Annotations), but it doesn't work. I've added css class with style:

<style type="text/css">
        .tooltipA {
            cursor: pointer;
            position: absolute;
            display: inline;
            background: green;
            right: 0;
            margin-right: 105%;
        }
    </style>

and then 
var newopts = {
...
annotateClassName: "tooltip"
        }

And for hover effect I need to change mousemove function somehow?

Thanks!!
